src/main/java/io/vertx/codegen/type/ReflectTypeInternal.java

```diff
@@ -0,0 +1,83 @@
+package io.vertx.codegen.type;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.AnnotatedParameterizedType;
+import java.lang.reflect.AnnotatedType;
+import java.lang.reflect.Method;
+import java.lang.reflect.ParameterizedType;
+
+import javax.annotation.processing.ProcessingEnvironment;
+import javax.lang.model.element.ExecutableElement;
+
+import io.vertx.codegen.Helper;
+import io.vertx.codegen.type.TypeUse.TypeInternal;
+
+class ReflectTypeInternal implements TypeInternal {
+
+  public static final TypeUse.TypeInternalProvider PROVIDER = new TypeUse.TypeInternalProvider() {
+    private Method getMethod(ProcessingEnvironment env, ExecutableElement methodElt) {
+      Method methodRef = Helper.getReflectMethod(Thread.currentThread().getContextClassLoader(), methodElt);
```

Comment:
The usage of the TCCL can be null. Does it check this case and use the current classloader?

Reply:
that's actually in code generation (javac), not at vertx runtime so the context is quite defined here